From 575a3aa37c0a1a5f993cb49d1477fb3a0ff05152 Mon Sep 17 00:00:00 2001
From: Mat <mail@mathias.is>
Date: Thu, 10 Aug 2023 05:03:31 +0300
Subject: [PATCH] game: Only create a single timer

A new, redundant timer was created each time tile pairs were removed
or hints were shown, leading to excessive resource usage over time.

Fixes #27
Fixes #39

Part-of: <https://gitlab.gnome.org/GNOME/gnome-mahjongg/-/merge_requests/36>
---
 src/game.vala | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/game.vala b/src/game.vala
index 87d0fc5..8f1ae27 100644
--- a/src/game.vala
+++ b/src/game.vala
@@ -445,8 +445,9 @@ public class Game : Object
 
     private void start_clock ()
     {
-        if (clock == null)
-            clock = new Timer ();
+        if (clock != null)
+            return;
+        clock = new Timer ();
         timeout_cb ();
     }
 
-- 
GitLab

